src/main/python/systemds/script_building/script.py

```diff
@@ -79,7 +79,34 @@ def execute(self) -> JavaObject:
                                                             _list_to_java_array([self.out_var_name]))
             for (name, input_node) in self.inputs.items():
                 input_node.pass_python_data_to_prepared_script(gateway.jvm, name, self.prepared_script)
-        return self.prepared_script.executeScript()
+
+            if lineage:
+                connection.setLineage(True)
+
+        ret = self.prepared_script.executeScript()
+        if lineage:
+            print(self.prepared_script.getLineageTrace(self.out_var_name))
+
+        return ret
+
+    def getlineage(self) -> str:
+        gateway = get_gateway()
+        entry_point = gateway.entry_point
+        if self.prepared_script is None:
+            input_names = self.inputs.keys()
+            connection = entry_point.getConnection()
+            self.prepared_script = connection.prepareScript(self.dml_script,
+                                                            _list_to_java_array(input_names),
+                                                            _list_to_java_array([self.out_var_name]))
+            for (name, input_node) in self.inputs.items():
+                input_node.pass_python_data_to_prepared_script(gateway.jvm, name, self.prepared_script)
+
+            connection.setLineage(True)
+
+        ret = self.prepared_script.executeScript()
+        lineage = self.prepared_script.getLineageTrace(self.out_var_name)
+        return lineage
+        
```

Comment:
Could we maybe add this to `.executeScript()`, since the code seems to be the almost the same

Reply:
This looks the same because to get the lineage you need to execute the script. But from an user's point, they do different things.

Reply:
Will the user ever have direct access to the Script? I think currently Script is a non public object (we might change that in the future).

Reply:
This routine is not accessed by the users, they call getLineageTrace which internally calls this. But you are right that as now compute too can return lineage trace, that can be reused. Will do it later.